diesel/src/connection/transaction_manager.rs

```diff
@@ -371,6 +378,12 @@ where
             ),
             None => return Err(Error::NotInTransaction),
         };
+        let depth = transaction_state
+            .transaction_depth()
+            .expect("We know that we are in a transaction here");
+        conn.instrumentation().on_connection_event(
```

Comment:
maybe we could avoid the additional dynamic dispatches in the transaction manager when no instrumentation is enabled by having `Connection::instrumentation` return `Option<&mut dyn Instrumentation>` ?
I'm not sure that would be worth the added complexity though 😅

Reply:
I would prefer keeping the additional dynamic dispatches in the transaction manager. I don't think it's particular important to optimize away a few nanoseconds there. On the other hand the current design allows to keep the `Connection::instrumentation` function as opaque a possible, which hopefully makes it easier to incorporate future changes there.